### PR TITLE
feat: Order detail: User can see more shipping info for their order [PX-4210]

### PR DESCRIPTION
**Pull request by Serge0n**
The type of this PR is: **Feature**
Design 🎨  : [Figma](https://www.figma.com/file/J5bIQj4a0l7wYAHnhbVGTp/%5BCollector%5D-Checkout-Shipping?node-id=2826%3A0) 

This PR resolves [PX-4210]

### Description

This PR is adding a new `TrackOrderSection` to the OrderDetails page and allows users to see tracking and timing info about their order. Also, this PR returns correct bold text for sections titles.

We need to doublecheck backend fields for `Shipped on`, `Estimated Delivery` and `Delivered on` fields in this section. So please, correct this if it's wrong and let me know:point_down:
1. If CommerceShip:
- Shipped on -> `createdAt` on `CommerceFulfillment`
- Estimated Delivery -> `estimatedDelivery` on `CommerceFulfillment`
- if status "delivered" then `Delivered on` -> :question::question::question: on `CommerceFulfillment` (hidding now)
2. If CommerceShipArta:
- Shipped on -> `deliveryStart` on `CommerceShipment`
- Estimated Delivery -> `estimatedDeliveryWindow` on `CommerceShipment`
- if status "delivered" then `Delivered on` -> `deliveryEnd` on `CommerceShipment`

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Order Details:  added TrackOrderSection - Serge0n

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

**VIDEO:clapper: :**

https://user-images.githubusercontent.com/55637696/134161145-4d65152b-3639-41d8-8aef-2343d37f5621.mov

[PX-4210]: https://artsyproduct.atlassian.net/browse/PX-4210

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Order Details:  added TrackOrderSection - Serge0n)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 40ada6018181b0ce125b33fa3b0d2efad8cd0175
</p>

File: src/lib/Scenes/OrderHistory/OrderDetails/Components/TrackOrderSection.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const [fulfillment] = extractNodes(fulfillments)
+  const { estimatedDelivery, createdAt } = fulfillment || {}
+  const trackingUrl = getTrackingUrl(lineItem)
+  const orderStatus = getOrderStatus(section.state as OrderState, lineItem)
+  const deliveredStatus = orderStatus === "delivered"
+
+  return (
+    <Flex flexDirection="row" justifyContent="space-between">
+      <Flex>
+        <Text testID="orderStatus" variant="text" style={{ textTransform: "capitalize" }}>
```
Comment (MounirDhahri) [inline src/lib/Scenes/OrderHistory/OrderDetails/Components/TrackOrderSection.tsx:31]: **nitpick:** I strongly suggest to use @dimatretyak https://github.com/artsy/to-title-case here :)

File: src/lib/utils/getTrackingUrl.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+import { TrackOrderSection_section } from "__generated__/TrackOrderSection_section.graphql"
 
 export type OrderHistoryRowLineItem = NonNullable<
   NonNullable<NonNullable<OrderHistoryRow_order["lineItems"]>["edges"]>[0]
 >["node"]
 
-export function getTrackingUrl(orderLineItem: OrderHistoryRowLineItem): string | undefined {
+export type OrderDetailsTrackOrderSectionLineItem = NonNullable<
+  NonNullable<NonNullable<TrackOrderSection_section["lineItems"]>["edges"]>[0]
+>["node"]
```
Comment (MounirDhahri) [inline src/lib/utils/getTrackingUrl.ts:10]: **praise:** 🙌

**Review by MounirDhahri — APPROVED** [review]
Nice one, LGTM🎉 I left a small comment but it not a blocker at all

**Comment by MounirDhahri** [comment]
Side note: I think there is something wrong with your git config. I am not getting your git handle to show up properly

Screenshot 2021-09-21 at 18 24 33

**Review by ansor4 — APPROVED** [review]
sweet!

**Pull request merged**